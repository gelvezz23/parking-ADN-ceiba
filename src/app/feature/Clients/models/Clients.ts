export interface Clients {
  id: number;
  day: string;
  date: string;
  responsable: string;
  idResponsable: string;
  licensePlate: string;
}
