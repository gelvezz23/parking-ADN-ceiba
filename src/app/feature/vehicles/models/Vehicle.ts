export interface Vehicle {
  id: number;
  day: string;
  date: string;
  slot: string;
  isActive: boolean;
  responsable: string;
  type: string;
  idResponsable: string;
  licensePlate: string;
}
