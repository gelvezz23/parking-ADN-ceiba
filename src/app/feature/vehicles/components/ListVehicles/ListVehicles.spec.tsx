import * as React from 'react';
import { RenderResult, render } from '@testing-library/react';
import { SinonStub, stub } from 'sinon';
import { ListVehicles } from './index';

import { Vehicle } from '../../models/Vehicle';
import { BrowserRouter as Router } from 'react-router-dom';

describe('test List Clients', () => {
  let componentWrapper: RenderResult;
  let componentProps: React.ComponentProps<typeof ListVehicles> & {
    onClickMarkOut: SinonStub;
    vehicles: Array<Vehicle>;
    limitSlots: number;
  };
  const vehicles = [
    {
      id: 1,
      day: 'Domingo',
      date: '2021-03-29T03:17:22.363Z',
      isActive: true,
      slot: 'M1',
      responsable: 'Carlos',
      type: 'Moto',
      idResponsable: '1090503315',
      licensePlate: 'XZY88',
    },
    {
      id: 2,
      day: 'Domingo',
      date: '2021-03-29T03:22:17.907Z',
      isActive: true,
      slot: 'M1',
      responsable: 'Carlos',
      type: 'Carro',
      idResponsable: '1090503315',
      licensePlate: 'XZY888',
    },
    {
      id: 4,
      day: 'Lunes',
      date: '2021-03-29T05:36:57.884Z',
      isActive: true,
      slot: 'W1',
      responsable: 'Carlos',
      type: 'Weight',
      idResponsable: '1090503315',
      licensePlate: 'XZY889',
    },
  ];

  beforeEach(() => {
    componentProps = {
      vehicles: vehicles,
      limitSlots: 22,
      onClickMarkOut: stub(),
    };
    componentWrapper = render(<ListVehicles {...componentProps} />);
  });

  it('should match snapshot', () => {
    expect(componentWrapper.container).toMatchSnapshot();
  });

  it('should render 9 p', () => {
    const elem = componentWrapper.container;
    const p = elem.querySelectorAll('p');
    const expected = 9;
    expect(p.length).toBe(expected);
  });

  it('should render 3 images', () => {
    const elem = componentWrapper.container;
    const img = elem.querySelectorAll('img');
    const expected = 3;
    expect(img.length).toBe(expected);
  });

  it('should render 3 titles', () => {
    const elem = componentWrapper.container;
    const h5 = elem.querySelectorAll('h5');
    const expected = 3;
    expect(h5.length).toBe(expected);
  });
});
