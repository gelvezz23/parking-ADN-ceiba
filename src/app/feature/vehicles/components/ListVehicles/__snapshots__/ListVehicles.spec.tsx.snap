// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test List Clients should match snapshot 1`] = `
<div>
  <div
    class="fadeInDown"
  >
    <h5>
      Motos
    </h5>
    <div
      class="sc-AxiKw bReNwz"
    >
      <div
        class="animate__heartBeat"
      >
        <div
          class="sc-AxjAm hWztmq"
        >
          <div
            class="sc-AxirZ jEmvie"
          >
            <img
              alt="Imagen de Moto"
              src="moto.png"
              style="width: 50px;"
            />
            <p>
              Carlos
            </p>
            <p>
              XZY88
            </p>
            <p>
              Moto
            </p>
            <button
              class="sc-AxhCb fNxkEz"
            >
              <span
                aria-labelledby="trash"
                role="img"
              >
                salida ðŸ•‘
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="fadeInDown"
  >
    <h5>
      Carros 
    </h5>
    <div
      class="sc-AxiKw bReNwz"
    >
      <div
        class="animate__heartBeat"
      >
        <div
          class="sc-AxjAm hWztmq"
        >
          <div
            class="sc-AxirZ jEmvie"
          >
            <img
              alt="Imagen de Carro"
              src="car.png"
              style="width: 50px;"
            />
            <p>
              Carlos
            </p>
            <p>
              XZY888
            </p>
            <p>
              Carro
            </p>
            <button
              class="sc-AxhCb fNxkEz"
            >
              <span
                aria-labelledby="trash"
                role="img"
              >
                salida ðŸ•‘
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="fadeInDown"
  >
    <h5>
      Vehiculos Pesados 
    </h5>
    <div
      class="sc-AxiKw bReNwz"
    >
      <div
        class="animate__heartBeat"
      >
        <div
          class="sc-AxjAm hWztmq"
        >
          <div
            class="sc-AxirZ jEmvie"
          >
            <img
              alt="Imagen de vehiculo pesado"
              src="weight.png"
              style="width: 50px;"
            />
            <p>
              Carlos
            </p>
            <p>
              XZY889
            </p>
            <p>
              Weight
            </p>
            <button
              class="sc-AxhCb fNxkEz"
            >
              <span
                aria-labelledby="trash"
                role="img"
              >
                salida ðŸ•‘
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
